strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/io/LazyInputStream.java

```diff
@@ -0,0 +1,132 @@
+package org.carlspring.strongbox.io;
+
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * @author sbespalov
```

Comment:
Would you mind adding some comments/documentation of when these two "lazy" streams should be used and what's special about them? Thanks!

Reply:
done